### PR TITLE
Refactor node chart to be production ready

### PR DESCRIPTION
**Pull request by tompizmor**
This refactor includes the following features:

- Supports ingress (disabled by default) 
- Supports ingress TLS termination (the secret with the TLS certificate must exist before deploying the chart or use [kube-lego](https://github.com/kubernetes/charts/tree/master/stable/kube-lego) )
- The number of replicas of the application is configurable via values.
- It uses a minimal node image to run the application (`bitnami/node:TAG-prod`)
- If your application requires a database you can provision a production-ready one running in the Azure cloud using k8s service catalog and the Open Service Broker for Azure.

### CODE REVIEW

File: incubator/node/README.md [hunk 1]
```diff
@@ -83,3 +95,58 @@ The [Bitnami Node](https://github.com/bitnami/bitnami-docker-node) image stores
...
+
+1. Install Service Catalog in your Kubernetes cluster following [this instructions](https://kubernetes.io/docs/tasks/service-catalog/install-service-catalog-using-helm/)
+2. Install the Open Service Broker for Azure in your Kubernetes cluster following [this instructions](https://github.com/Azure/helm-charts/tree/master/open-service-broker-azure)
+3. Deploy the helm chart:
+
+    ```
+    $ helm install --name node-app --set externalDatabase.azure.enabled=true bitnami-incubator/node
+    ```
+
+Setting `externalDatabase.azure.enabled` to `true` makes the chart to create a `ServiceInstance` and a `ServiceBinding` kubernetes resources.
```
Comment (arapulido) [inline incubator/node/README.md]: Mmm, is this what we wanted to do? I thought the idea was to make it more generic (not only for Azure), but have instructions on how to use the sevice catalog to create an instance in Azure, and connect it using the generic externalDatabase values.

Comment (tompizmor) [reply incubator/node/README.md]: We thought to do it in this way because the ServiceInstance and ServiceBinding resources are automatically created for the use. Make it more generic means that the user will need to write and create these resources on its own before deploys the chart.  
Also, the idea was to consume the database credentials from the Secret instead of passing them as standard helm values.

Comment (prydonius) [reply incubator/node/README.md]: So I think what we should do here is consume the secret, and not create the ServiceInstance and ServiceBinding. We can assume that those have been created, out of band, and just take in the result of that binding (i.e. the secret).

The main reason for this is that ServiceInstances usually require a range of parameters that are very specific to the cloud platform. Whilst we could easily make those parameters configurable via chart values, I think it's better to not create a direct dependency on the ServiceInstance.

This means that another scenario we can support is when you already have a ServiceInstance (and maybe also a ServiceBinding) and want to use that instead of have the chart provision one for you. With the current approach, it's not possible to use an existing ServiceInstance.

We should document the steps needed to create the ServiceInstance and ServiceBinding, either by having example YAML files or using the svcat CLI. The chart should then take in a secret (`--set externalDatabase.secretName=<name>`) and we should document what keys we expect that secret to have (for now, it should be whatever the OSBA ServiceBinding creates for us).

WDYT?

File: incubator/node/templates/deployment.yaml [hunk 2]
```diff
@@ -1,22 +1,23 @@
...
-        app: {{ template "fullname" . }}
+        app: {{ template "node.name" . }}
+        release: {{ .Release.Name | quote }}
     spec:
       initContainers:
       - name: git-clone-app
-        image: {{ .Values.image }}
+        image: {{ .Values.devImage }}
         imagePullPolicy: {{ .Values.imagePullPolicy }}
         command: [ '/bin/sh', '-c' , 'git clone {{ .Values.repository }} /app && git checkout {{ .Values.revision }}']
```
[the remark this answers is not in the document]
Comment (tompizmor) [reply incubator/node/templates/deployment.yaml]: I tried using the gitRepo volume but it clones the repo in ${mountPoint}/${repoName} while I want to always clone in `/app` to not depends on the name. I am using no the alpine/git image

File: incubator/node/templates/deployment.yaml [hunk 3]
```diff
@@ -1,22 +1,23 @@
 apiVersion: extensions/v1beta1
 kind: Deployment
 metadata:
-  name: {{ template "fullname" . }}
+  name: {{ template "node.fullname" . }}
   labels:
-    app: {{ template "fullname" . }}
+    app: {{ template "node.name" . }}
     chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
```
[the remark this answers is not in the document]
Comment (tompizmor) [reply incubator/node/templates/deployment.yaml]: good catch!

File: incubator/externaldb-helper/Chart.yaml [hunk 4]
```diff
@@ -0,0 +1,13 @@
+name: externaldb-helper
```
[the remark this answers is not in the document]
Comment (tompizmor) [reply incubator/externaldb-helper/Chart.yaml]: @prydonius @sameersbn thoughts?

File: incubator/node/README.md [hunk 5]
```diff
@@ -97,7 +97,7 @@ The [Bitnami Node](https://github.com/bitnami/bitnami-docker-node) image stores
 Persistent Volume Claims are used to keep the data across deployments. This is known to work in GCE, AWS, and minikube.
 See the [Configuration](#configuration) section to configure the PVC or to disable persistence.
 
-## Set an Ingress
+## Set an Ingress controller
```
Comment (sameersbn) [inline incubator/node/README.md]: `Set` -> `Set up`

File: incubator/node/templates/deployment.yaml [hunk 6]
```diff
@@ -17,9 +17,9 @@ spec:
     spec:
       initContainers:
       - name: git-clone-app
-        image: {{ .Values.devImage }}
+        image: {{ .Values.gitImage }}
         imagePullPolicy: {{ .Values.imagePullPolicy }}
-        command: [ '/bin/sh', '-c' , 'git clone {{ .Values.repository }} /app && git checkout {{ .Values.revision }}']
+        command: ['git', 'clone', '{{ .Values.repository }}', '-b', '{{ .Values.revision }}', '/app']
```
Comment (sameersbn) [inline incubator/node/templates/deployment.yaml]: `-b` argument for `git clone` will not work for revisions. It should either be a tag or branch.

File: incubator/node/README.md [hunk 7]
```diff
@@ -47,27 +47,40 @@ The command removes all the Kubernetes components associated with the chart and
...
-| `resources`                   | CPU/Memory resource requests/limits | Memory: `512Mi`, CPU: `300m`                          |
+|              Parameter                  |            Description                                    |                        Default                            |
+|-----------------------------------------|-----------------------------------------------------------|-----------------------------------------------------------|
+| `image`                                 | Node image                                                | `bitnami/node:{VERSION}`                                  |
+| `gitImage`                              | Image used for initContainers                             | `alpine/git`                                              |
+| `imagePullPolicy`                       | Image pull policy                                         | `IfNotPresent`                                            |
+| `repository`                            | Repo of the application                                   | `https://github.com/jbianquetti-nami/sample-mean.git`     |
```
Comment (sameersbn) [inline incubator/node/README.md]: should be https://github.com/bitnami/sample-mean.git

File: incubator/node/README.md [hunk 8]
```diff
@@ -83,3 +96,98 @@ The [Bitnami Node](https://github.com/bitnami/bitnami-docker-node) image stores
...
+      app: mongodb
+  spec:
+    clusterServiceClassExternalName: azure-cosmos-mongo-db
+    clusterServicePlanExternalName: mongo-db
+    parameters:
+      location: YOUR_AZURE_LOCATION
+      resourceGroup: mongodb-k8s-service-catalog
+      sslEnforcement: disabled
+      firewallStartIPAddress: 0.0.0.0
+      firewallEndIPAddress: 255.255.255.255
```
Comment (sameersbn) [inline incubator/node/README.md]: the firewall specification has changed for osba, see https://github.com/Azure/helm-charts/commit/75ab2232e80c1469ad262faa8039ac64262c5c8c#diff-d04a73ad2755189100f39bf28c84a22e for the necessary changes.

Comment (tompizmor) [reply incubator/node/README.md]: thanks!

**Comment by tompizmor** [comment]
@prydonius @sameersbn I moved the bitnami-common chart to this other PR https://github.com/bitnami/charts/pull/639 . Until that chart is not merged the ci will fail for this PRL

File: incubator/node/README.md [hunk 9]
```diff
@@ -5,14 +5,14 @@
 ## TL;DR;
```
Comment (sameersbn) [inline incubator/node/README.md]: `TL;DR;` -> `TL;DR`

File: incubator/node/README.md [hunk 10]
```diff
@@ -47,30 +47,47 @@ The command removes all the Kubernetes components associated with the chart and
 
 The following tables lists the configurable parameters of the Node chart and their default values.
```
Comment (sameersbn) [inline incubator/node/README.md]: `The following tables lists` -> `The following table lists`

File: incubator/node/README.md [hunk 11]
```diff
@@ -47,30 +47,47 @@ The command removes all the Kubernetes components associated with the chart and
...
-| `resources`                   | CPU/Memory resource requests/limits | Memory: `512Mi`, CPU: `300m`                              |
+|              Parameter                  |            Description                                    |                        Default                            |
+|-----------------------------------------|-----------------------------------------------------------|-----------------------------------------------------------|
+| `image.registry`                        | NodeJS image registry                                     | `docker.io`                                               |
+| `image.repository`                      | NodeJS Image name                                         | `bitnami/node`                                            |
+| `image.tag`                             | NodeJS Image tag                                          | `{VERSION}`                                               |
+| `image.pullPolicy`                      | NodeJS image pull policy                                  | `Always` if `imageTag` is `latest`, else `IfNotPresent`   |
```
Comment (sameersbn) [inline incubator/node/README.md]: default value is `IfNotPresent`.

File: incubator/node/README.md [hunk 12]
```diff
@@ -47,30 +47,47 @@ The command removes all the Kubernetes components associated with the chart and
...
+| `gitImage.registry`                     | Git image registry                                        | `docker.io`                                               |
+| `gitImage.repository`                   | Git Image name                                            | `alpine/git`                                              |
+| `gitImage.tag`                          | Git Image tag                                             | `latest`                                                  |
+| `gitImage.pullPolicy`                   | Git image pull policy                                     | `Always` if `imageTag` is `latest`, else `IfNotPresent`   |
+| `gitImage`                              | Image used for initContainers                             | `alpine/git`                                              |
+| `repository`                            | Repo of the application                                   | `https://github.com/bitnami/sample-mean.git`              |
+| `revision`                              | Revision  to checkout                                     | `master`                                                  |
```
Comment (sameersbn) [inline incubator/node/README.md]: remove whitespaces in `Revision to checkout`

File: incubator/node/README.md [hunk 13]
```diff
@@ -47,30 +47,47 @@ The command removes all the Kubernetes components associated with the chart and
...
+| `replicas`                              | Number of replicas for the application                    | `3`                                                       |
```
Comment (sameersbn) [inline incubator/node/README.md]: defaults to `1`

File: incubator/node/README.md [hunk 14]
```diff
@@ -47,30 +47,47 @@ The command removes all the Kubernetes components associated with the chart and
...
 ```console
 $ helm install --name my-release \
-  --set repository=https://github.com/jbianquetti-nami/simple-node-app.git,mariadb.mariadbRootPassword=secretpassword \
-    incubator/node
+  --set repository=https://github.com/jbianquetti-nami/simple-node-app.git,replicas=2 \
+    bitnami-incubator/node
 ```
 
-The above command clones the remote git  repository to the `/app/` directory  of the container. Additionally it sets the MariaDB `root` user password to `secretpassword`.
+The above command clones the remote git repository to the `/app/` directory  of the container. Additionally it sets the number or `replicas` to `2`.
```
Comment (sameersbn) [inline incubator/node/README.md]: number ~or~of replicas

File: incubator/node/README.md [hunk 15]
```diff
@@ -86,3 +103,100 @@ The [Bitnami Node](https://github.com/bitnami/bitnami-docker-node) image stores
 
 Persistent Volume Claims are used to keep the data across deployments. This is known to work in GCE, AWS, and minikube.
 See the [Configuration](#configuration) section to configure the PVC or to disable persistence.
+
+## Set an Ingress controller
```
Comment (sameersbn) [inline incubator/node/README.md]: `Set an` -> `Set up an`
